apps/api_web/lib/api_web/rate_limiter/rate_limiter_concurrent.ex

```diff
@@ -0,0 +1,170 @@
+defmodule ApiWeb.RateLimiter.RateLimiterConcurrent do
+  @moduledoc """
+  Rate limits a user's concurrent connections based on their API key or by their IP address if no
+  API key is provided. Split by static and event-stream requests.
+  """
+
+  use GenServer
+  require Logger
+  alias ApiWeb.RateLimiter.Memcache.Supervisor
+
+  @rate_limit_concurrent_config Application.compile_env!(:api_web, :rate_limiter_concurrent)
+
+  def start_link([]), do: GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+
+  def init(_) do
+    {:ok, %{uuid: UUID.uuid1()}}
+  end
+
+  defp lookup(%ApiWeb.User{} = user, event_stream?) do
+    type = if event_stream?, do: "event_stream", else: "static"
+    key = "concurrent_#{user.id}_#{type}"
+
+    {
+      type,
+      key
+    }
+  end
+
+  def get_pid_key(pid) do
+    sub_key = pid |> :erlang.pid_to_list() |> to_string
+    get_uuid() <> sub_key
+  end
+
+  defp get_uuid do
+    {:ok, uuid} = GenServer.call(__MODULE__, :get_uuid)
```

Comment:
suggestion: this is creating another bottleneck, as the server can only respond to one `get_uuid` request at a time. Some options would be:
- a protected ETS table based on the name of the server
- a `:persistent_term` value set when the server starts

Reply:
I have updated to use `:persistent_term`.